iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -194,6 +210,54 @@ public boolean canProvideBasicStatistics() {
     return stats;
   }
 
+  public boolean addDynamicSplitPruningEdge(org.apache.hadoop.hive.ql.metadata.Table table,
+      ExprNodeDesc syntheticFilterPredicate) {
+    try {
+      Collection<String> partitionColumns = ((HiveIcebergSerDe) table.getDeserializer()).partitionColumns();
+      if (partitionColumns.size() > 0) {
+        // Collect the column names from the predicate
+        Set<String> filterColumns = Sets.newHashSet();
+        columns(syntheticFilterPredicate, filterColumns);
+
+        // While Iceberg could handle multiple columns the current pruning only able to handle filters for a
+        // single column. We keep the logic below to handle multiple columns so if pruning is available on executor
+        // side the we can easily adapt to it as well.
+        if (filterColumns.size() > 1) {
```

Comment:
If we know that we don't support multiple column filtering, wouldn't be possible to get a rough estimate of the filter size ( is >=1) before collecting every column name?

Reply:
I do not understand this. Could you please ellaborate?

Thanks,
Peter

Reply:
We collect every column name in the filterColumns set through the columns() method. That method is traversing every node recursively, so it might be time-consuming.  After that, the size of the set is validated, and if it's greater than 1, return false. 
Can we introduce some logic, to fail fast, without the need of traversing every node? I'm just thinking aloud, I don't know whether it is feasible or not.